spec.html

```diff
@@ -35569,6 +36178,61 @@ <h1>ClassRanges</h1>
         <emu-alg>
           1. Return the CharSet that is the result of evaluating |NonemptyClassRanges|.
         </emu-alg>
+        <p>The production <emu-grammar>ClassRanges :: ClassContents</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Return the CharSet that is the result of evaluating |ClassContents|.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-classstrings">
+        <h1>ClassStrings</h1>
+        <p>The production <emu-grammar>ClassStrings :: `(` ClassString MoreClassStrings? `)`</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |ClassString| to obtain a string _s_.
+          1. Let A be the CharSet that contains the one string _s_.
+          1. If |MoreClassStrings| is present, then
+            1. Evaluate |MoreClassStrings| to obtain a CharSet _B_.
+            1. Return the union of CharSets _A_ and _B_.
+          1. Return _A_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-moreclassstrings">
+        <h1>MoreClassStrings</h1>
+        <p>The production <emu-grammar>MoreClassStrings :: `|` ClassString MoreClassStrings?</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |ClassString| to obtain a string _s_.
+          1. Let _A_ be the CharSet that contains the one string _s_.
+          1. If |MoreClassStrings| is present, then
+            1. Evaluate |MoreClassStrings| to obtain a CharSet _B_.
+            1. Return the union of CharSets _A_ and _B_.
+          1. Return _A_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-classstring">
+        <h1>ClassString</h1>
+        <p>The production <emu-grammar>ClassString :: [empty]</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Return the empty String.
+        </emu-alg>
+        <p>The production <emu-grammar>ClassString :: NonEmptyClassString</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |NonEmptyClassString| to obtain a string _s_.
+          1. Return _s_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-nonemptyclassstring">
+        <h1>NonEmptyClassString</h1>
+        <p>The production <emu-grammar>NonEmptyClassString :: ClassCharacter NonEmptyClassString?</emu-grammar> evaluates as follows:</p>
+        <emu-alg>
+          1. Evaluate |ClassCharacter| to obtain the single-character string _s1_.
```

Comment:
Evaluating |ClassCharacter| returns a single-element CharSet, not the element itself.